datacube/utils/documents.py

```diff
@@ -426,6 +428,7 @@ def __init__(self, doc: dict[str, Any], sources_path=None) -> None:
         self._sources = None
         self._doc_uuid: UUID | None = None
 
+    # FIXME: despite the type signature, this returns a Mapping.
     @property
```

Comment:
Probably a hangover from my general rule of thumb: typehint input args as loosely as possible, and return args as tightly as possible.

I presume simply declaring Mapping here causes flow-on type-checking issues too extensive or tangled to be addressed in this PR.